pkgs/tools/security/quill/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib, rustPlatform, fetchFromGitHub, openssl, pkg-config, protobuf, which }:
+
+rustPlatform.buildRustPackage rec {
+  name = "quill-${version}";
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "dfinity";
+    repo = "quill";
+    rev = "v${version}";
+    sha256 = "02ga2xkdxs36mfr4lv43cy6wkf27c28bdkzfkp3az5jvyk17mkfr";
+  };
+
+  ic = fetchFromGitHub {
+    owner = "dfinity";
+    repo = "ic";
+    rev = "779549eccfcf61ac702dfc2ee6d76ffdc2db1f7f";
+    sha256 = "1r31d5hab7k1n60a7y8fw79fjgfq04cgj9krwa6r9z4isi3919v6";
+  };
+
+  registry = "file://local-registry";
+
+  preBuild = ''
+    export REGISTRY_TRANSPORT_PROTO_INCLUDES=${ic}/rs/registry/transport/proto
+    export IC_BASE_TYPES_PROTO_INCLUDES=${ic}/rs/types/base_types/proto
+    export IC_PROTOBUF_PROTO_INCLUDES=${ic}/rs/protobuf/def
+    export IC_NNS_COMMON_PROTO_INCLUDES=${ic}/rs/nns/common/proto
+    export PROTOC=$(which protoc)
```

Comment:
```suggestion
    export PROTOC=${protobuf}/bin/protoc
```

Reply:
this is not the same thing, because I need to protoc to be the native form,

when cross compiling, if you do ${protoc} you get the binary compiled for the target architecture

maybe there is a more idomatic way to do this

Reply:
Then we need to use buildPackages.